common/flatpak-utils.c

```diff
@@ -1140,6 +1140,213 @@ flatpak_compare_ref (const char *ref1, const char *ref2)
+              return NULL;
+            }
+        }
+      else if (c == '*')
+        {
+          g_string_append (regexp, "[.\\-_a-zA-Z0-9]*");
+        }
+      else if (c == '.')
+        {
+          g_string_append (regexp, "\\.");
+        }
+      else if (g_ascii_isalnum (c) || c == '-' || c == '_')
+        {
+          g_string_append_c (regexp, c);
+        }
+      else
+        {
+          flatpak_fail_error (error, FLATPAK_ERROR_INVALID_DATA, _("Invalid glob character '%c'"), c);
+          return NULL;
+        }
+    }
+
+  while (parts < 3)
+    {
+      parts++;
+      g_string_append (regexp, "/[.\\-_a-zA-Z0-9]*");
+    }
+
+  return g_string_free (g_steal_pointer (&regexp), FALSE);
+}
+
+gboolean
+flatpak_parse_filters (const char *data,
+                       GRegex **allow_refs_out,
+                       GRegex **deny_refs_out,
+                       GError **error)
+{
+  g_auto(GStrv) lines = NULL;
+  int i;
+  g_autoptr(GString) allow_regexp = g_string_new ("^(");
+  g_autoptr(GString) deny_regexp = g_string_new ("^(");
+  gboolean has_allow = FALSE;
+  gboolean has_deny = FALSE;
+  g_autoptr(GRegex) allow_refs = NULL;
+  g_autoptr(GRegex) deny_refs = NULL;
+
+  lines = g_strsplit (data, "\n", -1);
+  for (i = 0; lines[i] != NULL; i++)
+    {
+      char *line = lines[i];
+      char *comment, *command;
+
+      /* Ignore shell-style comments */
+      comment = strchr (line, '#');
+      if (comment != NULL)
+        *comment = 0;
+
+      command = line_get_word (&line);
+      /* Ignore empty lines */
+      if (command == NULL)
+        continue;
+
+      if (strcmp (command, "allow") == 0 || strcmp (command, "deny") == 0)
+        {
+          char *glob, *next, *ref_regexp;
+          GString *command_regexp;
+          gboolean *has_type = NULL;
+
+          glob = line_get_word (&line);
+          if (glob == NULL)
+            return flatpak_fail_error (error, FLATPAK_ERROR_INVALID_DATA, _("Missing glob on line %d"), i + 1);
+
+          next = line_get_word (&line);
+          if (next != NULL)
+            return flatpak_fail_error (error, FLATPAK_ERROR_INVALID_DATA, _("Trailing text on line %d"), i + 1);
+
+          ref_regexp = glob_to_regexp (glob, error);
+          if (ref_regexp == NULL)
+            return glnx_prefix_error (error, _("on line %d"), i + 1);
+
+          if (strcmp (command, "allow") == 0)
+            {
+              command_regexp = allow_regexp;
+              has_type = &has_allow;
+            }
+          else
+            {
+              command_regexp = deny_regexp;
+              has_type = &has_deny;
+            }
+
+          if (*has_type)
+            g_string_append (command_regexp, "|");
+          else
+            *has_type = TRUE;
+
+          g_string_append (command_regexp, ref_regexp);
+        }
+      else
+        {
+          return flatpak_fail_error (error, FLATPAK_ERROR_INVALID_DATA, _("Invalid op '%s' on line %d"), command, i + 1);
```

Comment:
Is 'op' a great word here ? Maybe good enough, if the docs consistently talk about it like that.

Reply:
Well, its pretty obvious what is meant as we include the word, but yeah, maybe not ideal wording